src/mame/barcrest/mpu4.cpp

```diff
@@ -924,7 +929,15 @@ uint8_t mpu4_state::pia_ic5_portb_r()
 WRITE_LINE_MEMBER(mpu4_state::pia_ic5_ca2_w)
 {
 	LOG(("%s: IC5 PIA Write CA2 (Serial Tx) %2x\n",machine().describe_context(),state));
-	m_dataport->write_txd(state);
+	if (m_dataport_loopback)
+	{
+		m_serial_data = state;
+		m_pia4->cb1_w(state);
+	}
+	else
+	{
+		m_dataport->write_txd(state);		
+	}
 }
```

Comment:
Please get rid of this member entirely and just hook up the IC5 CA2 output to either the PIA4 CB1 input or the data logger directly in machine configuration.

Reply:
Unfortunately, the fact that the edge detect is traced to an additional input port (IC4, port B, pin 7) means I had to keep serial detect around, I couldn't find an easy way of duplicating that signal.